README
NAME

    Dist::Zilla::Plugin::AutoPrereqs::Perl::Critic - automatically extract
    Perl::Critic policy prereqs

VERSION

    Version 0.006

SYNOPSIS

      # in dist.ini:
      [AutoPrereqs::Perl::Critic]
      critic_config        = .perlcriticrc  ; defaults to not specify a profile
      phase                = develop        ; default
      type                 = requires       ; default
      remove_core_policies = 1              ; default

DESCRIPTION

    This plugin will add Perl::Critic and all policies used by it, in the
    installed version, as distribution prerequisites.

 critic_config

    By default no policy is specified which lets Perl::Critic find the
    config itself. This defaults to .perlcriticrc in the current directory.

 phase

    By default, the dependencies are added to the develop phase. This can
    be changed to every valid phase.

 remove_core_policies

    By default, policies that are included in the latest Perl::Critic
    distribution are not added as dependency. This can be changed by
    setting remove_core_policies to 0.

    Note: Perl::Critic itself is always added as dependency which will come
    with the core policies.

    Note: This feature needs HTTP access to cpanmetadb.plackperl.org.
    Please disable this feature if you're system cannot access that server.

    Note: To reduce network traffic and remove the delay caused by the
    network access the cache file .perlcritic_package.yml is generated. You
    can either add this file to your .gitignore file or add it to Git. It
    will be updated as soon as the system runs a newer version of
    Perl::Critic then the one that is mentioned in the cache file.

 type

    By default, the dependencies are added as type requires. This can be
    changed to every valid phase.

SUPPORT

 Bugs / Feature Requests

    Please report any bugs or feature requests through the issue tracker at
    https://github.com/skirmess/Dist-Zilla-Plugin-AutoPrereqs-Perl-Critic/issues.
    You will be notified automatically of any progress on your issue.

 Source Code

    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    https://github.com/skirmess/Dist-Zilla-Plugin-AutoPrereqs-Perl-Critic

      git clone https://github.com/skirmess/Dist-Zilla-Plugin-AutoPrereqs-Perl-Critic.git

AUTHOR

    Sven Kirmess <sven.kirmess@kzone.ch>

SEE ALSO

    Dist::Zilla::Plugin::AutoPrereqs, Perl::Critic, Test::Perl::Critic



Changes
Revision history for Dist-Zilla-Plugin-AutoPrereqs-Perl-Critic

{{$NEXT}}

  - Switch license to the ISC license
  - Removed minimal dependency on Moose 2.1203
  - Code cleanup - moved all 'use' before 'has'

0.006  2018-04-02 22:51:55 UTC

  - Test for success of HTTP::Tiny instead of status
  - Distribution cleanup
  - Bumped dependency on Moose to 2.1203

0.005  2017-10-21 09:53:51 UTC

  - Create a .perlcritic_package.yml cache file to reduce network access

0.004  2017-09-21 22:39:29 UTC

  - Added a test for no Internet access
  - Require no Internet access for tests

0.003  2017-09-14 19:21:17 UTC

  - Added VERSION section to pod
  - Removed AutoPrereqs::Perl::Critic from dist.ini because we are part of
    the Author::SKIRMESS bundle

0.002  2017-08-30 21:02:54 UTC

  - Remove dependencies or core plugins
  - Added option remove_core_policies to also depend on core plugins

0.001  2017-08-29 22:22:52 UTC

  - Initial release


t/lib/Local/HTTP/Tiny/Mock.pm
# vim: ts=4 sts=4 sw=4 et: syntax=perl
#
# Copyright (c) 2017-2022 Sven Kirmess
#
# Permission to use, copy, modify, and distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

use 5.006;
use strict;
use warnings;

package Local::HTTP::Tiny::Mock;

our $VERSION = '0.001';

sub get_200 {

    # perl -e 'use HTTP::Tiny; use Data::Dumper; print Dumper(HTTP::Tiny->new->get(q{http://cpanmetadb.plackperl.org/v1.0/package/Perl::Critic}));'

    return sub {

        return {
            'url'     => 'http://cpanmetadb.plackperl.org/v1.0/package/Perl::Critic',
            'content' => <<'EOF',
---
distfile: P/PE/PETDANCE/Perl-Critic-1.130.tar.gz
provides:
  Perl::Critic: 1.130
  Perl::Critic::Annotation: 1.130
  Perl::Critic::Command: 1.130
  Perl::Critic::Config: 1.130
  Perl::Critic::Document: 1.130
  Perl::Critic::Exception: 1.130
  Perl::Critic::Exception::AggregateConfiguration: 1.130
  Perl::Critic::Exception::Configuration: 1.130
  Perl::Critic::Exception::Configuration::Generic: 1.130
  Perl::Critic::Exception::Configuration::NonExistentPolicy: 1.130
  Perl::Critic::Exception::Configuration::Option: 1.130
  Perl::Critic::Exception::Configuration::Option::Global: 1.130
  Perl::Critic::Exception::Configuration::Option::Global::ExtraParameter: 1.130
  Perl::Critic::Exception::Configuration::Option::Global::ParameterValue: 1.130
  Perl::Critic::Exception::Configuration::Option::Policy: 1.130
  Perl::Critic::Exception::Configuration::Option::Policy::ExtraParameter: 1.130
  Perl::Critic::Exception::Configuration::Option::Policy::ParameterValue: 1.130
  Perl::Critic::Exception::Fatal: 1.130
  Perl::Critic::Exception::Fatal::Generic: 1.130
  Perl::Critic::Exception::Fatal::Internal: 1.130
  Perl::Critic::Exception::Fatal::PolicyDefinition: 1.130
  Perl::Critic::Exception::IO: 1.130
  Perl::Critic::Exception::Parse: 1.130
  Perl::Critic::OptionsProcessor: 1.130
  Perl::Critic::Policy: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitBooleanGrep: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitComplexMappings: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitLvalueSubstr: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitReverseSortBlock: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitSleepViaSelect: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitStringyEval: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitStringySplit: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitUniversalCan: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitUniversalIsa: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitUselessTopic: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitVoidGrep: 1.130
  Perl::Critic::Policy::BuiltinFunctions::ProhibitVoidMap: 1.130
  Perl::Critic::Policy::BuiltinFunctions::RequireBlockGrep: 1.130
  Perl::Critic::Policy::BuiltinFunctions::RequireBlockMap: 1.130
  Perl::Critic::Policy::BuiltinFunctions::RequireGlobFunction: 1.130
  Perl::Critic::Policy::BuiltinFunctions::RequireSimpleSortBlock: 1.130
  Perl::Critic::Policy::ClassHierarchies::ProhibitAutoloading: 1.130
  Perl::Critic::Policy::ClassHierarchies::ProhibitExplicitISA: 1.130
  Perl::Critic::Policy::ClassHierarchies::ProhibitOneArgBless: 1.130
  Perl::Critic::Policy::CodeLayout::ProhibitHardTabs: 1.130
  Perl::Critic::Policy::CodeLayout::ProhibitParensWithBuiltins: 1.130
  Perl::Critic::Policy::CodeLayout::ProhibitQuotedWordLists: 1.130
  Perl::Critic::Policy::CodeLayout::ProhibitTrailingWhitespace: 1.130
  Perl::Critic::Policy::CodeLayout::RequireConsistentNewlines: 1.130
  Perl::Critic::Policy::CodeLayout::RequireTidyCode: 1.130
  Perl::Critic::Policy::CodeLayout::RequireTrailingCommas: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitCStyleForLoops: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitCascadingIfElse: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitDeepNests: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitLabelsWithSpecialBlockNames: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitMutatingListFunctions: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitNegativeExpressionsInUnlessAndUntilConditions: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitPostfixControls: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitUnlessBlocks: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitUnreachableCode: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitUntilBlocks: 1.130
  Perl::Critic::Policy::ControlStructures::ProhibitYadaOperator: 1.130
  Perl::Critic::Policy::Documentation::PodSpelling: 1.130
  Perl::Critic::Policy::Documentation::RequirePackageMatchesPodName: 1.130
  Perl::Critic::Policy::Documentation::RequirePodAtEnd: 1.130
  Perl::Critic::Policy::Documentation::RequirePodLinksIncludeText: 1.130
  Perl::Critic::Policy::Documentation::RequirePodSections: 1.130
  Perl::Critic::Policy::ErrorHandling::RequireCarping: 1.130
  Perl::Critic::Policy::ErrorHandling::RequireCheckingReturnValueOfEval: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitBacktickOperators: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitBarewordFileHandles: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitExplicitStdin: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitInteractiveTest: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitJoinedReadline: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitOneArgSelect: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitReadlineInForLoop: 1.130
  Perl::Critic::Policy::InputOutput::ProhibitTwoArgOpen: 1.130
  Perl::Critic::Policy::InputOutput::RequireBracedFileHandleWithPrint: 1.130
  Perl::Critic::Policy::InputOutput::RequireBriefOpen: 1.130
  Perl::Critic::Policy::InputOutput::RequireCheckedClose: 1.130
  Perl::Critic::Policy::InputOutput::RequireCheckedOpen: 1.130
  Perl::Critic::Policy::InputOutput::RequireCheckedSyscalls: 1.130
  Perl::Critic::Policy::InputOutput::RequireEncodingWithUTF8Layer: 1.130
  Perl::Critic::Policy::Miscellanea::ProhibitFormats: 1.130
  Perl::Critic::Policy::Miscellanea::ProhibitTies: 1.130
  Perl::Critic::Policy::Miscellanea::ProhibitUnrestrictedNoCritic: 1.130
  Perl::Critic::Policy::Miscellanea::ProhibitUselessNoCritic: 1.130
  Perl::Critic::Policy::Modules::ProhibitAutomaticExportation: 1.130
  Perl::Critic::Policy::Modules::ProhibitConditionalUseStatements: 1.130
  Perl::Critic::Policy::Modules::ProhibitEvilModules: 1.130
  Perl::Critic::Policy::Modules::ProhibitExcessMainComplexity: 1.130
  Perl::Critic::Policy::Modules::ProhibitMultiplePackages: 1.130
  Perl::Critic::Policy::Modules::RequireBarewordIncludes: 1.130
  Perl::Critic::Policy::Modules::RequireEndWithOne: 1.130
  Perl::Critic::Policy::Modules::RequireExplicitPackage: 1.130
  Perl::Critic::Policy::Modules::RequireFilenameMatchesPackage: 1.130
  Perl::Critic::Policy::Modules::RequireNoMatchVarsWithUseEnglish: 1.130
  Perl::Critic::Policy::Modules::RequireVersionVar: 1.130
  Perl::Critic::Policy::NamingConventions::Capitalization: 1.130
  Perl::Critic::Policy::NamingConventions::ProhibitAmbiguousNames: 1.130
  Perl::Critic::Policy::Objects::ProhibitIndirectSyntax: 1.130
  Perl::Critic::Policy::References::ProhibitDoubleSigils: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitCaptureWithoutTest: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitComplexRegexes: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitEnumeratedClasses: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitEscapedMetacharacters: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitFixedStringMatches: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitSingleCharAlternation: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitUnusedCapture: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitUnusualDelimiters: 1.130
  Perl::Critic::Policy::RegularExpressions::ProhibitUselessTopic: 1.130
  Perl::Critic::Policy::RegularExpressions::RequireBracesForMultiline: 1.130
  Perl::Critic::Policy::RegularExpressions::RequireDotMatchAnything: 1.130
  Perl::Critic::Policy::RegularExpressions::RequireExtendedFormatting: 1.130
  Perl::Critic::Policy::RegularExpressions::RequireLineBoundaryMatching: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitAmpersandSigils: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitBuiltinHomonyms: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitExcessComplexity: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitExplicitReturnUndef: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitManyArgs: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitNestedSubs: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitReturnSort: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitSubroutinePrototypes: 1.130
  Perl::Critic::Policy::Subroutines::ProhibitUnusedPrivateSubroutines: 1.130
  Perl::Critic::Policy::Subroutines::ProtectPrivateSubs: 1.130
  Perl::Critic::Policy::Subroutines::RequireArgUnpacking: 1.130
  Perl::Critic::Policy::Subroutines::RequireFinalReturn: 1.130
  Perl::Critic::Policy::TestingAndDebugging::ProhibitNoStrict: 1.130
  Perl::Critic::Policy::TestingAndDebugging::ProhibitNoWarnings: 1.130
  Perl::Critic::Policy::TestingAndDebugging::ProhibitProlongedStrictureOverride: 1.130
  Perl::Critic::Policy::TestingAndDebugging::RequireTestLabels: 1.130
  Perl::Critic::Policy::TestingAndDebugging::RequireUseStrict: 1.130
  Perl::Critic::Policy::TestingAndDebugging::RequireUseWarnings: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitCommaSeparatedStatements: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitComplexVersion: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitConstantPragma: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitEmptyQuotes: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitEscapedCharacters: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitImplicitNewlines: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitLeadingZeros: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitLongChainsOfMethodCalls: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitMagicNumbers: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitMismatchedOperators: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitMixedBooleanOperators: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitNoisyQuotes: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitQuotesAsQuotelikeOperatorDelimiters: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitSpecialLiteralHeredocTerminator: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::ProhibitVersionStrings: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::RequireConstantVersion: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::RequireInterpolationOfMetachars: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::RequireNumberSeparators: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::RequireQuotedHeredocTerminator: 1.130
  Perl::Critic::Policy::ValuesAndExpressions::RequireUpperCaseHeredocTerminator: 1.130
  Perl::Critic::Policy::Variables::ProhibitAugmentedAssignmentInDeclaration: 1.130
  Perl::Critic::Policy::Variables::ProhibitConditionalDeclarations: 1.130
  Perl::Critic::Policy::Variables::ProhibitEvilVariables: 1.130
  Perl::Critic::Policy::Variables::ProhibitLocalVars: 1.130
  Perl::Critic::Policy::Variables::ProhibitMatchVars: 1.130
  Perl::Critic::Policy::Variables::ProhibitPackageVars: 1.130
  Perl::Critic::Policy::Variables::ProhibitPerl4PackageNames: 1.130
  Perl::Critic::Policy::Variables::ProhibitPunctuationVars: 1.130
  Perl::Critic::Policy::Variables::ProhibitReusedNames: 1.130
  Perl::Critic::Policy::Variables::ProhibitUnusedVariables: 1.130
  Perl::Critic::Policy::Variables::ProtectPrivateVars: 1.130
  Perl::Critic::Policy::Variables::RequireInitializationForLocalVars: 1.130
  Perl::Critic::Policy::Variables::RequireLexicalLoopIterators: 1.130
  Perl::Critic::Policy::Variables::RequireLocalizedPunctuationVars: 1.130
  Perl::Critic::Policy::Variables::RequireNegativeIndices: 1.130
  Perl::Critic::PolicyConfig: 1.130
  Perl::Critic::PolicyFactory: 1.130
  Perl::Critic::PolicyListing: 1.130
  Perl::Critic::PolicyParameter: 1.130
  Perl::Critic::PolicyParameter::Behavior: 1.130
  Perl::Critic::PolicyParameter::Behavior::Boolean: 1.130
  Perl::Critic::PolicyParameter::Behavior::Enumeration: 1.130
  Perl::Critic::PolicyParameter::Behavior::Integer: 1.130
  Perl::Critic::PolicyParameter::Behavior::String: 1.130
  Perl::Critic::PolicyParameter::Behavior::StringList: 1.130
  Perl::Critic::ProfilePrototype: 1.130
  Perl::Critic::Statistics: 1.130
  Perl::Critic::TestUtils: 1.130
  Perl::Critic::Theme: 1.130
  Perl::Critic::ThemeListing: 1.130
  Perl::Critic::UserProfile: 1.130
  Perl::Critic::Utils: 1.130
  Perl::Critic::Utils::Constants: 1.130
  Perl::Critic::Utils::DataConversion: 1.130
  Perl::Critic::Utils::McCabe: 1.130
  Perl::Critic::Utils::POD: 1.130
  Perl::Critic::Utils::POD::ParseInteriorSequence: 1.130
  Perl::Critic::Utils::PPI: 1.130
  Perl::Critic::Utils::Perl: 1.130
  Perl::Critic::Violation: 1.130
  Test::Perl::Critic::Policy: 1.130
version: 1.130
EOF
            'protocol' => 'HTTP/1.1',
            'headers'  => {
                'server'         => 'nginx/1.6.3',
                'cache-control'  => 'max-age=1800',
                'content-type'   => 'text/yaml',
                'x-timer'        => 'S1506018172.065497,VS0,VE2',
                'x-cache-hits'   => '1, 1',
                'content-length' => '12653',
                'date'           => 'Thu, 21 Sep 2017 18:22:52 GMT',
                'connection'     => 'keep-alive',
                'x-cache'        => 'HIT, HIT',
                'age'            => '130393',
                'via'            => [
                    '1.1 varnish',
                    '1.1 varnish',
                ],
                'accept-ranges'       => 'bytes',
                'fastly-debug-digest' => '5bd2af97315cd0addda0fd4657bbd4cf437331d546f05b2adba5573b6c88c030',
                'x-runtime'           => '0.085299',
                'x-served-by'         => 'cache-sjc3147-SJC, cache-hhn1546-HHN',
            },
            'status'  => '200',
            'success' => 1,
            'reason'  => 'OK',
        };
    };
}

sub get_404 {

    # perl -e 'use HTTP::Tiny; use Data::Dumper; print Dumper(HTTP::Tiny->new->get(q{http://cpanmetadb.plackperl.org/v1.0/package/Perl::Critic_does_not_exist}));'

    return sub {
        return {
            'headers' => {
                'accept-ranges' => 'bytes',
                'x-cache'       => 'MISS, MISS',
                'date'          => 'Thu, 21 Sep 2017 19:16:05 GMT',
                'server'        => 'nginx/1.6.3',
                'content-type'  => 'text/plain',
                'via'           => [
                    '1.1 varnish',
                    '1.1 varnish',
                ],
                'fastly-debug-digest' => 'abe8f8abc3c33a054b261732aa0cb98d62f1ca227a5aafd2fa53981ccd3d1f62',
                'connection'          => 'keep-alive',
                'x-served-by'         => 'cache-sjc3151-SJC, cache-hhn1538-HHN',
                'x-timer'             => 'S1506021365.113918,VS0,VE208',
                'x-cache-hits'        => '0, 0',
                'x-runtime'           => '0.046718',
                'content-length'      => '10',
                'age'                 => '0',
            },
            'protocol' => 'HTTP/1.1',
            'content'  => "Not found\n",
            'success'  => q{},
            'status'   => '404',
            'url'      => 'http://cpanmetadb.plackperl.org/v1.0/package/Perl::Critic_does_not_exist',
            'reason'   => 'Not Found',
        };
    };
}

1;


Makefile.PL
use 5.006;
use strict;
use warnings;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
    ABSTRACT           => 'automatically extract Perl::Critic policy prereqs',
    AUTHOR             => 'Sven Kirmess <sven.kirmess@kzone.ch>',
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => 0,
    },
    DISTNAME         => 'Dist-Zilla-Plugin-AutoPrereqs-Perl-Critic',
    LICENSE          => 'open_source',
    MIN_PERL_VERSION => '5.006',
    NAME             => 'Dist::Zilla::Plugin::AutoPrereqs::Perl::Critic',
    PREREQ_PM        => {
        'CPAN::Meta::YAML'                => 0,
        'CPAN::Version'                   => 0,
        'Dist::Zilla::Role::PrereqSource' => 0,
        'HTTP::Tiny'                      => 0,
        Moose                             => 0,
        'Moose::Util::TypeConstraints'    => 0,
        'Path::Tiny'                      => 0,
        'Perl::Critic'                    => 0,
        'namespace::autoclean'            => 0,
        strict                            => 0,
        warnings                          => 0,
    },
    TEST_REQUIRES => {
        'Test::DZil'       => 0,
        'Test::Fatal'      => 0,
        'Test::MockModule' => '0.14',
        'Test::More'       => '0.88',
        lib                => 0,
    },
    VERSION => '0.006',
    test    => {
        TESTS => 't/*.t',
    },
);

my %FallbackPrereqs = (
    'CPAN::Meta::YAML'                => 0,
    'CPAN::Version'                   => 0,
    'Dist::Zilla::Role::PrereqSource' => 0,
    'HTTP::Tiny'                      => 0,
    Moose                             => 0,
    'Moose::Util::TypeConstraints'    => 0,
    'Path::Tiny'                      => 0,
    'Perl::Critic'                    => 0,
    'Test::DZil'                      => 0,
    'Test::Fatal'                     => 0,
    'Test::MockModule'                => '0.14',
    'Test::More'                      => '0.88',
    lib                               => 0,
    'namespace::autoclean'            => 0,
    strict                            => 0,
    warnings                          => 0,
);

if ( !eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
    delete $WriteMakefileArgs{TEST_REQUIRES};
    delete $WriteMakefileArgs{BUILD_REQUIRES};
    $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

if ( !eval { ExtUtils::MakeMaker->VERSION(6.52) } ) {
    delete $WriteMakefileArgs{CONFIGURE_REQUIRES};
}

WriteMakefile(%WriteMakefileArgs);

# vim: ts=4 sts=4 sw=4 et: syntax=perl
